src/bionics.cpp

```diff
@@ -520,13 +521,12 @@ void npc::check_or_use_weapon_cbm( const bionic_id &cbm_id )
     bionic &bio = ( *my_bionics )[index];
 
     if( bio.info().has_flag( json_flag_BIONIC_GUN ) ) {
-        if( !bio.info().fake_item.is_valid() ) {
-            debugmsg( "tried to activate gun bionic \"%s\" with invalid fake_item \"%s\"", bio.info().id.str(),
-                      bio.info().fake_item.str() );
+        if( !bio.info().fake_item ) {
```

Comment:
`is_valid` is different from `is_empty` in that a non-empty item id can still be invalid (i.e. not pointing to any item definition).

Reply:
Yes, that's why at the validation step any invalid fake_item IDs are nulled (set to cata::nullopt). After validation there shouldn't be any CBMs pointing to an invalid fake_item ID, simplifying the checks as validity and emptiness are checked already on load.